tests/zfs-tests/tests/functional/cli_root/zfs_mount/zfs_mount_test_race.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/ksh
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2019 by Tomohiro Kusumi. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# DESCRIPTION:
+# Verify parallel mount result is consistent.
+# See github.com/zfsonlinux/zfs/issues/8833 for details.
+#
+# STRATEGY:
+# 1. Create pools and filesystems.
+# 2. Set same mount point for >1 datasets.
+# 3. Unmount all datasets.
+# 4. Mount all datasets.
+# 5. Unmount all datasets.
+# 6. Mount all datasets.
+#
+
+verify_runnable "both"
+
+log_note "Verify parallel mount result is consistent"
+
+TMPDIR=${TMPDIR:-$TEST_BASE_DIR}
+MNTPT=$TMPDIR/zfs_mount_test_race_mntpt
+
+DISK1="$TMPDIR/zfs_mount_test_race_disk1"
+TESTPOOL1=zfs_mount_test_race_tp1
+TESTFS1=zfs_mount_test_race_tf1
+
+DISK2="$TMPDIR/zfs_mount_test_race_disk2"
+TESTPOOL2=zfs_mount_test_race_tp2
+TESTFS2=zfs_mount_test_race_tf2
+
+log_must zfs unmount -a # unmount zfs mounts from previous tests
+log_must rm -rf $MNTPT
+log_must rm -rf /$TESTPOOL1
+log_must rm -rf /$TESTPOOL2
+
+function cleanup
+{
+	zpool destroy $TESTPOOL1
+	zpool destroy $TESTPOOL2
+	rm -rf $MNTPT
+	rm -rf /$TESTPOOL1
+	rm -rf /$TESTPOOL2
+	log_must zfs mount -a # remount unmounted zfs mounts
+}
+log_onexit cleanup
+
+log_note "Testing github.com/zfsonlinux/zfs/issues/8833"
```

Comment:
Since the issue was referenced in the above comment, I think a human readable explaination would be better here.  How about the following based on your existing `log_pass` message.

```
 log_note " log_pass "Verify parallel mount ordering is consistent"
```